seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/source/JdbcSourceReader.java

```diff
@@ -60,25 +60,23 @@ public void close() throws IOException {
     @Override
     @SuppressWarnings("magicnumber")
     public void pollNext(Collector<SeaTunnelRow> output) throws Exception {
-        synchronized (output.getCheckpointLock()) {
```

Comment:
Removing secondary locks causes snapshot state corruption

Reply:
Because the lock is acquired inside the output.collect() method, removing the lock in the pollNext() method will not result in snapshot corruption.

Reply:
The `snapshotState` method and `pollNext` work together to operate splits. Removing the lock will cause the split to be lost.

Reply:
ok, i got it, thx.